### PR TITLE
CC | action: Automate CC payload release

### PR DESCRIPTION
**Pull request by fidencio**
Let's create a GitHub action to automate the Kata Containers payload generation for the Confidential Containers project.

This GitHub action builds the artefacts (in parallel), merges them into a single tarball, generates the payload with the resulting tarball, and uploads the payload to the Confidential Containers quay.io.

It expects the tags to be used to be in the `CC-x.y.z` format, with x, y, and z being numbers.

Fixes: #5330

Signed-off-by: Fabiano Fidêncio <fabiano.fidencio@intel.com>

### CODE REVIEW

**Review by stevenhorsman — APPROVED** [review]
LGTM. What are your thoughts about pushing a `latest` tag at the same time. Not necessarily for this PR, but in general?

**Comment by fidencio** [comment]
> LGTM. What are your thoughts about pushing a `latest` tag at the same time. Not necessarily for this PR, but in general?

Not really comfortable doing that right now, as we cannot just add a `latest` there, but rather tag as `kata-containers-latest`, considering we share the namespace with enclave-cc.  I'd rather switch to passing the tag used instead.  For instance, tagginc `kata-containers-0.2.0` there (which could also cause confusion, considering the version wouldn't match the Kata Containers project version).

Let me sleep on this suggestion for a day or two. :-)

**Comment by stevenhorsman** [comment]
> > LGTM. What are your thoughts about pushing a `latest` tag at the same time. Not necessarily for this PR, but in general?
> 
> Not really comfortable doing that right now, as we cannot just add a `latest` there, but rather tag as `kata-containers-latest`, considering we share the namespace with enclave-cc. I'd rather switch to passing the tag used instead. For instance, tagginc `kata-containers-0.2.0` there (which could also cause confusion, considering the version wouldn't match the Kata Containers project version).
> 
> Let me sleep on this suggestion for a day or two. :-)

Yeah, it's an interesting one. I guess the question comes down to who is consuming the payload Image. At the moment I have two 'users':
- The operator in it's cc runtime - I think this fine as we do/plan to do a separate operator release which wraps this, so having it be manually updated there seems fine for now
- The cloud-api-adaptor payload image - this is potentially the more interesting one as it is is based on our kata payload image e.g: https://github.com/confidential-containers/cloud-api-adaptor/blob/445d72562267c61583c4e923332c92c6689819de/install/runtime-payload/Dockerfile#L2. However given that #5461 provides a CI 'latest' build that we could potentially use for operator CI and we are only planning to release every six weeks then manually adding it to the mental checklist to update seems fine to me. As I've chatted with Pradipta, longer term we might want to consider adding the kata-remote config to the payload image and then remove the CAA payload entirely, but that's a question for the future once peer pods is a proper part of the coco release.
Both of these users are likely to be done by community members, so if there aren't any 'outsiders' picking it up then needing to understand the sha of the released payload image shouldn't be too difficult and is maybe something we could add to release notes?

**TL;DR: ** - I think I agree - with #5461 an the current use cases, a release latest tag is not important at the moment. 
Thanks for indulging my brain turning this one over!

**Comment by fidencio** [comment]
Removed the `do-not-merge` label as I fixed @wainersm's comments from #5461 here as well.

**Review by wainersm — APPROVED** [review]
